hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileBlock.java

```diff
@@ -1709,6 +1700,9 @@ protected HFileBlock readBlockDataInternal(FSDataInputStream is, long offset,
         if (verifyChecksum && !validateChecksum(offset, curBlock, hdrSize)) {
           return null;
         }
+        // remove checksum from buffer now that it's verified
+        int sizeWithoutChecksum = curBlock.getInt(Header.ON_DISK_DATA_SIZE_WITH_HEADER_INDEX);
+        curBlock = curBlock.limit(sizeWithoutChecksum);
```

Comment:
This is just a no op? curBlock.limit just returns curBlock itself?

Reply:
ok. I personally prefer to use the returned value in cases like this, in case the implementation changes. but i'm happy to not do that if it's more generally preferred. i just pushed a change to remove that.